src/app/auth.service.ts

```diff
@@ -0,0 +1,96 @@
+import { Injectable, OnInit } from '@angular/core';
+import { ApiService } from './api.service';
+import { LoginResponse } from './api.model';
+import { Observable, Subject } from 'rxjs';
+import 'rxjs/add/observable/of';
+
+
+@Injectable({
+    providedIn: 'root'
+})
+export class AuthService {
+
+    private isLogged = new Subject<boolean>();
+    private currentTimeoutFunction: any;
+    private TOKEN_KEY = 'auth_token';
+    private TOKEN_EXP_KEY = 'auth_exp';
+
+    constructor(
+        public readonly apiService: ApiService,
+    ) {
+        if (this.authTokenIsPresent()) {
+            if (this.authTokenHasExpired()) {
+                this.refreshToken();
+            } else {
+                this.isLogged.next(true);
+            }
+        } else {
+            this.isLogged.next(false);
+        }
+    }
+
+    public login(email: string, password: string): Observable<LoginResponse> {
+        const response = this.apiService.login(email, password);
+        response.subscribe(
+            data => this.saveToken(data),
+            err => console.log(err)
```

Comment:
potremmo mettere `console.error()` dato che e' un errore
```suggestion
            err => console.error(err)
```